### PR TITLE
Veoss 552

### PR DESCRIPTION
**Pull request by oliwiaKot**

### CODE REVIEW

File: app/config/Module.scala [hunk 1]
```diff
@@ -19,13 +19,21 @@ package config
...
-import java.time.{Clock, ZoneOffset}
+import java.time.{Clock, ZoneId, ZoneOffset, ZonedDateTime}
 
 class Module extends AbstractModule {
 
   override def configure(): Unit = {
     bind(classOf[DataRequiredAction]).to(classOf[DataRequiredActionImpl]).asEagerSingleton()
-    bind(classOf[Clock]).toInstance(Clock.systemDefaultZone.withZone(ZoneOffset.UTC))
+//    bind(classOf[Clock]).toInstance(Clock.systemDefaultZone.withZone(ZoneOffset.UTC))
+    bind(classOf[Clock]).toInstance(
```
Comment (darbyo) [inline app/config/Module.scala:29]: Can we revert this back to the original?

File: app/controllers/corrections/CorrectPreviousReturnController.scala [hunk 2]
```diff
@@ -56,10 +60,19 @@ class CorrectPreviousReturnController @Inject()(
...
 
         value =>
-          for {
-            updatedAnswers <- Future.fromTry(request.userAnswers.set(CorrectPreviousReturnPage, value))
-            _              <- cc.sessionRepository.set(updatedAnswers)
-          } yield Redirect(CorrectPreviousReturnPage.navigate(mode, updatedAnswers))
-      )
+      for {
+        updatedAnswers <- Future.fromTry(request.userAnswers.set(CorrectPreviousReturnPage, value))
+        periods <- returnStatusConnector.listStatuses(request.registration.commencementDate)
```
Comment (darbyo) [inline app/controllers/corrections/CorrectPreviousReturnController.scala:65]: Can we move the connector call to inside the yield as if it fails then our session repo wont get the updatedUserAnswers

File: app/views/corrections/CorrectionReturnPeriodView.scala.html [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
 @layout(pageTitle = title(form, messages("correctionReturnPeriod.title"))) {
 
-    @formHelper(action = controllers.corrections.routes.CorrectionReturnPeriodController.onSubmit(mode, period), 'autoComplete -> "off") {
+    @formHelper(action = controllers.corrections.routes.CorrectionReturnPeriodController.onSubmit(mode, period, Index(0)), 'autoComplete -> "off") {
```
Comment (DevAndyroo) [inline app/views/corrections/CorrectionReturnPeriodView.scala.html:29]: Should the index be passed as a parameter?

File: app/config/Module.scala [hunk 4]
```diff
@@ -19,13 +19,21 @@ package config
...
+      Clock.fixed(
+        ZonedDateTime.of(
+          2022, 1, 1, 1, 0, 0, 0, ZoneOffset.UTC
+        ).toInstant,
+        ZoneId.of("Australia/Melbourne")
+      )
+    )
```
Comment (DevAndyroo) [inline app/config/Module.scala:36]: remove the mock clock

**Review by DevAndyroo — CHANGES_REQUESTED** [review]

File: app/controllers/corrections/CorrectPreviousReturnController.scala [hunk 5]
```diff
@@ -56,10 +60,19 @@ class CorrectPreviousReturnController @Inject()(
...
+        _ <- cc.sessionRepository.set(updatedAnswers)
+      } yield {
+        periods match {
+          case Right(periods) =>Redirect(CorrectPreviousReturnPage.navigate(mode, updatedAnswers, periods.size))
+          case Left(value) =>
+            logger.error(s"there was an error $value")
+            throw new Exception(value.toString)
```
Comment (darbyo) [inline app/controllers/corrections/CorrectPreviousReturnController.scala:72]: I think redirecting to journeyRecovery might be better here

**Review by darbyo — CHANGES_REQUESTED** [review]

File: app/controllers/corrections/CorrectPreviousReturnController.scala (hunk 2 above)
Comment (darbyo) [reply app/controllers/corrections/CorrectPreviousReturnController.scala:65]: OR just reorder them so periods is the last call would work

File: app/controllers/corrections/CorrectionCountryController.scala [hunk 6]
```diff
@@ -32,41 +35,67 @@ import scala.concurrent.{ExecutionContext, Future}
...
         value =>
-          for {
-            updatedAnswers <- Future.fromTry(request.userAnswers.set(CorrectionCountryPage, value))
-            _              <- cc.sessionRepository.set(updatedAnswers)
-          } yield Redirect(CorrectionCountryPage.navigate(mode, updatedAnswers))
+          request.userAnswers.get(CorrectionReturnPeriodPage(periodIndex)) match {
+            case Some(correctionPeriod) =>
+              for {
+                          updatedAnswers <- Future.fromTry(request.userAnswers.set(CorrectionCountryPage(periodIndex, countryIndex), value))
+                          vatReturnResult <- vatReturnConnector.get(correctionPeriod)
```
Comment (darbyo) [inline app/controllers/corrections/CorrectionCountryController.scala:80]: Can we swap again the connector call and repo call so userAnswers update regardless

Comment (oliwiaKot) [reply app/controllers/corrections/CorrectionCountryController.scala:80]: does it need to go inside the yield or just be after the repo call?

File: app/pages/corrections/CorrectionCountryPage.scala [hunk 7]
```diff
@@ -17,17 +17,26 @@
...
+  override def path: JsPath = JsPath \ corrections \ periodIndex.position \ correctionToCountry \ countryIndex.position \ toString
 
   override def toString: String = "correctionCountry"
 
-  override def navigateInNormalMode(answers: UserAnswers): Call =
-    routes.IndexController.onPageLoad()
+   def navigate(mode: Mode, answers: UserAnswers, countriesFromNi: Seq[Country], countriesFromEU: Seq[Country]): Call = {
+    answers.get(CorrectionCountryPage(periodIndex, countryIndex)) match {
+      case Some(country) =>
+        if(countriesFromNi.contains(country) || countriesFromEU.contains(country)){
```
Comment (darbyo) [inline app/pages/corrections/CorrectionCountryPage.scala:35]: We could combine the two lists where they are passed in to this method and only have one list to check if country exists... one to think about

Comment (oliwiaKot) [reply app/pages/corrections/CorrectionCountryPage.scala]: done now

File: app/controllers/corrections/CorrectionCountryController.scala (hunk 6 above)
Comment (oliwiaKot) [reply app/controllers/corrections/CorrectionCountryController.scala]: done

File: app/controllers/corrections/CorrectPreviousReturnController.scala (hunk 5 above)
Comment (oliwiaKot) [reply app/controllers/corrections/CorrectPreviousReturnController.scala]: done

File: app/controllers/corrections/CorrectPreviousReturnController.scala (hunk 2 above)
Comment (oliwiaKot) [reply app/controllers/corrections/CorrectPreviousReturnController.scala]: done, reordered them

File: app/config/Module.scala (hunk 1 above)
Comment (oliwiaKot) [reply app/config/Module.scala]: done now

File: app/config/Module.scala (hunk 4 above)
Comment (oliwiaKot) [reply app/config/Module.scala]: done now

File: app/views/corrections/CorrectionReturnPeriodView.scala.html (hunk 3 above)
Comment (oliwiaKot) [reply app/views/corrections/CorrectionReturnPeriodView.scala.html]: yes, thanks for spotting that, done now

**Comment by oliwiaKot** [comment]
retest this please

**Review by DevAndyroo — APPROVED** [review]

**Review by darbyo — APPROVED** [review]

**Pull request merged**